src/lxc/file_utils.c

```diff
@@ -147,7 +147,7 @@ ssize_t lxc_read_nointr_expect(int fd, void *buf, size_t count, const void *expe
 	ssize_t ret;
 
 	ret = lxc_read_nointr(fd, buf, count);
-	if (ret <= 0)
+	if (ret < 0)
```

Comment:
What, why?

Reply:
@brauner this function isn't used anywhere currently, so I've re-purposed it for my sysctl comparisons needs. Whilst doing so I noticed that this appears to have an edge case bug in it where if the result of `lxc_read_nointr` was 0 (and this would happen if the file being read is zero length) then it would return 0 without doing the buffer comparison - thus indicating that the the 2 buffers were equal even when they are potentially not.